GitHub - add issue and pull request templates

ethancrawford opened the pull request:
This change add a pull request template, as well as issue template forms for
Sonic Pi startup issues, runtime issues, compiling issues and feature requests.

ethancrawford committed:
GitHub - add issue and pull request templates

This change add a pull request template, as well as issue template forms for
Sonic Pi startup issues, runtime issues, compiling issues and feature requests.

ethancrawford commented:
Here are the issue template forms and a pull request template as discussed previously.
(This includes the issue template form for compilation issues as later suggested by @samaaron).

emlyn commented on .github/ISSUE_TEMPLATE/RUNNING-ISSUE-FORM.yml:
```diff
@@ -0,0 +1,117 @@
+        What version of Sonic Pi were you running?
+        (Was it a release version, Beta, or a version you compiled from source yourself?)
+      options:
+        - Self compiled
```
Will these lists need to be updated manually each time a new release is made? Might it be worth adding an "Other" option in case that ever gets forgotten?

ethancrawford replied:
Sure. Do you think there'd be any drawback to having it as a plain text input?

ethancrawford replied:
(The Sonic Pi version input that is)

emlyn replied:
Yes, I think plain text would be fine, and it would be less work from a maintenance point of view.

ethancrawford committed:
GitHub - change the Sonic Pi version selectors to text inputs

This will reduce maintenance burden when new releases or Betas are produced.

ethancrawford committed:
GitHub - tweak Feature Description description

ethancrawford commented:
For reference: I created a dummy repo with the above issue templates actually available for testing.
See: https://github.com/ethancrawford/test-templates/issues.

I'm calling this ready as soon as I have had agreement that my updates to the troubleshooting guide (now guide**s**) on my copy of the Sonic Pi wiki are good enough to merge to the upstream copy (and therefore I can re-link references to them in the above templates).

ethancrawford committed:
GitHub - tweak running issue error example

ethancrawford committed:
GitHub - update comments about code formatting

The phrasing was changed slightly, and the links to the GitHub docs for how to
use code formatting have been updated.

ethancrawford committed:
GitHub - tweak feature request Feature Description example

kbrock commented:
This is very slick.

For the projects I've worked on, we just added a few markdown comments and left it free form.

Glad to learn what is available.
(Not sure which one I prefer)
